sqlmesh/schedulers/airflow/operators/hwm_sensor.py

```diff
@@ -35,7 +35,10 @@ def __init__(
     def poke(self, context: Context) -> bool:
         dag_run = context["dag_run"]
 
-        target_snapshot = self._get_target_snapshot()
+        with util.scoped_state_sync() as state_sync:
+            target_snapshot = state_sync._get_snapshots([self.target_snapshot_info])[
```

Comment:
Why use the private method? We should just use the public one

Reply:
Yeah, but `scoped_state_sync()` returns a new instance every time, so there will be no cache.